### PR TITLE
bugfix: fix apply

### PR DESCRIPTION
**Pull request by Starnop**
### Describe what this PR does / why we need it

Fix the following two issues:
1. apply with mode `loadImage` will also launch the app
2. apply with no node scale will launch the apps  and do not distribute the image rootfs

### CODE REVIEW

File: cmd/sealer/cmd/cluster/apply.go [hunk 1]
```diff
@@ -127,12 +127,13 @@ func NewApplyCmd() *cobra.Command {
 					return applyClusterWithNew(cf, applyMode, imageEngine, imageSpec)
 				}
 
-				if err := applyClusterWithExisted(cf, client, imageEngine, imageSpec); err != nil {
+				updated, err := applyClusterWithExisted(cf, client, imageEngine, imageSpec)
```
Comment (kakaZhou719) [inline cmd/sealer/cmd/cluster/apply.go:130]: updated is too large to understand .  Or i thought should we move those run app logic  to applyClusterWithExisted.

File: cmd/sealer/cmd/cluster/apply.go [hunk 2]
```diff
@@ -127,12 +127,13 @@ func NewApplyCmd() *cobra.Command {
 					return applyClusterWithNew(cf, applyMode, imageEngine, imageSpec)
```
Comment (kakaZhou719) [inline cmd/sealer/cmd/cluster/apply.go:127]: applyClusterFileWithNew

File: cmd/sealer/cmd/cluster/apply.go (hunk 1 above)
Comment (kakaZhou719) [inline cmd/sealer/cmd/cluster/apply.go:130]: applyClusterFileWithExisted ?

File: cmd/sealer/cmd/cluster/apply.go [hunk 3]
```diff
@@ -127,12 +127,13 @@ func NewApplyCmd() *cobra.Command {
...
 
-				if err := applyClusterWithExisted(cf, client, imageEngine, imageSpec); err != nil {
+				updated, err := applyClusterWithExisted(cf, client, imageEngine, imageSpec)
+				if err != nil {
 					return err
 				}
 				// NOTE: we should continue to apply application after the cluster is applied successfully
 				// And it's not needed to prepare the app file repeatedly
-				ignorePrepareAppMaterials = true
+				ignorePrepareAppMaterials = updated
```
Comment (kakaZhou719) [inline cmd/sealer/cmd/cluster/apply.go:136]: SkipPrepareAppMaterials ?

File: cmd/sealer/cmd/cluster/apply.go (hunk 2 above)
Comment (Starnop) [reply cmd/sealer/cmd/cluster/apply.go:127]: in fact, it‘s just the part relevant to the cluster.

And as for what you said about the app, that's really unreasonable. And I have added a comment `TODO: decouple the cluster installation and application installation`, and we should fix it in a separate PR.

File: cmd/sealer/cmd/cluster/apply.go (hunk 1 above)
Comment (Starnop) [reply cmd/sealer/cmd/cluster/apply.go:130]: ditto

File: cmd/sealer/cmd/cluster/apply.go (hunk 3 above)
Comment (Starnop) [reply cmd/sealer/cmd/cluster/apply.go]: DONE

File: cmd/sealer/cmd/cluster/apply.go (hunk 1 above)
Comment (Starnop) [reply cmd/sealer/cmd/cluster/apply.go]: I have changed `updated` to `clusterUpdated`.

As with other comments, we should split run/apply to these steps:
1. distribute rootfs
2. apply/run cluster
3. app/run application

And then, something just like `clusterUpdated`, `skipPrepareAppMaterials ` will not be needed.

And we should do these code optimization ASAP.

**Review by kakaZhou719 — APPROVED** [review]
LGTM

**Pull request merged**